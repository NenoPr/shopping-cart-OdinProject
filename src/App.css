body::-webkit-scrollbar {
  display: block;
}

html {
  overflow-y: scroll;
  overflow: -moz-scrollbars-vertical; 
  overflow-y: scroll;
}

input[type=number]::-webkit-inner-spin-button,
input[type=number]::-webkit-outer-spin-button { 
	-webkit-appearance: none;
}

::-webkit-scrollbar {
  width: 1rem;
}

::-webkit-scrollbar-track {
  background: #696a7d;
}

::-webkit-scrollbar-thumb {
  background: #15233c;
  border-radius: 0.5rem;
}

::-webkit-scrollbar-thumb:hover {
  background: rgb(157, 175, 200);
}

.App {
  display: grid;
  grid-template: auto 14fr auto / 1fr;
  text-align: center;
  background-color: #282c34;
  min-height: 100vh;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.app-nav-bar {
  grid-area: 1 / 1 / 1 / 1;
  border-bottom: 0.1rem solid white;
  height: 100%;
  padding: 2rem;
}

.app-content-container {
  grid-area: 2 / 1 / 2 / 1;
  background-color: #262733;
  min-height: 100%;
}

.app-footer {
  grid-area: 3 / 1 / 3 / 1;
  border-top: 0.1rem solid white;
  padding: 1rem;
}
