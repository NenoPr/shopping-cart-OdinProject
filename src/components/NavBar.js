import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import uniqid from "uniqid";
import { ReactComponent as CartIcon } from "../shopping_cart.svg";
import "./styles/nav-bar.css";

function NavBar(appProps) {
  const linkStyle = {
    textDecoration: "none",
    color: "white",
  };

  console.log("appProps for NavBar", appProps);
  const [navCart, setNavCart] = useState(appProps.props);

  useEffect(() => {
    setNavCart(appProps.props);
  }, [appProps]);

  function displayCart() {
    let cartDisplayStatus = document.getElementById("cart");
    if (cartDisplayStatus.classList.contains("toggle-cart-display"))
      cartDisplayStatus.classList.remove("toggle-cart-display");
    else cartDisplayStatus.classList.add("toggle-cart-display");
  }

  return (
    <div className="nav-content">
      <div>Logo</div>
      <Link to={"/"} style={linkStyle}>
        <div className="nav-links">Home</div>
      </Link>
      <Link to={"/shop"} style={linkStyle}>
        <div className="nav-links">Shop</div>
      </Link>
      <Link to={"/about"} style={linkStyle}>
        <div className="nav-links">About</div>
      </Link>
      <div className="nav-links cart-svg" onClick={displayCart}>
        <CartIcon />
      </div>
      <div className="toggle-cart-display" id="cart">
        <div className="shopping-cart-container">
          <div className="close-shopping-cart" onClick={displayCart}>
            Close X
          </div>
          <div className="shopping-cart-headline">Shopping Cart</div>
          <div className="shopping-cart-article-list">
            {navCart.map((item) => (
              <div className="shopping-cart-item" key={uniqid()}>
                <div className="shopping-cart-product-name">
                  {item.productName}
                </div>
                <div className="shopping-cart-product-amount-container">
                  <div className="shopping-cart-product-amount-control">+</div>
                  <div className="shopping-cart-product-amount">
                    {item.amount}
                  </div>
                  <div className="shopping-cart-product-amount-control">-</div>
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
}

export default NavBar;
