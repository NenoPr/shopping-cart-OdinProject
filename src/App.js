import React, { useState } from "react";
import { HashRouter, Routes, Route } from "react-router-dom";
import NavBar from "./components/NavBar";
import Home from "./components/Home";
import Shop from "./components/Shop";
import About from "./components/About";
import ShopProductDetails from "./components/ShopProductDetails";
import Checkout from "./components/Checkout";
import AllProducts from "./components/ProductsData";
import PurchaseCompleted from "./components/PurchaseCompleted";
import "./App.css";

const App = () => {
  const [products, setProducts] = useState(AllProducts);
  const [cart, setCart] = useState([]);
  const [lastLocation, setLastLocation] = useState("")

  return (
    <HashRouter>
      <div className="App">
        <div className="app-nav-bar">
        <NavBar props={{cart, setCart}} />
        </div>
        <div className="app-content-container">
          <Routes>
            <Route path={"/"} element={<Home />} />
            <Route path={"/about"} element={<About />} />
            <Route path={"/shop"} element={<Shop items={{products, cart, setCart, setLastLocation}} />} />
            <Route path={`/shop/:categories`} element={<Shop items={{products, cart, setCart, lastLocation, setLastLocation}} />} />
            <Route path={`/shop/:categories/:product`} element={<ShopProductDetails props={{lastLocation, cart, setCart}}/>} />
            <Route path={"/checkout"} element={<Checkout props={{cart, setCart}}/>} />
            <Route path={"/purchase-completed"} element={<PurchaseCompleted props={{cart,setCart}} />} />
          </Routes>
        </div>
        <footer className="app-footer">Footer Content</footer>
      </div>
    </HashRouter>
  );
};

export default App;
