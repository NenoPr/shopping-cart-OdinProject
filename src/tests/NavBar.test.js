import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../App";
import NavBar from "../components/NavBar"

it("renders", () => {
    const {asFragment} = render(<App />)
    expect(asFragment()).toMatchSnapshot();
})

test('renders Home Link', () => {
  render(<NavBar />);
  const linkElement = screen.getByText(/Home/i);
  expect(linkElement).toBeInTheDocument();
});

