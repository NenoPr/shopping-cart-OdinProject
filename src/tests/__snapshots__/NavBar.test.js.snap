// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="app-nav-bar"
    >
      <div
        class="nav-content"
      >
        <div>
          Logo
        </div>
        <a
          class="nav-links"
          href="#/"
        >
          <div>
            Home
          </div>
        </a>
        <a
          class="nav-links"
          href="#/shop"
        >
          <div>
            Shop
          </div>
        </a>
        <a
          class="nav-links"
          href="#/about"
        >
          <div>
            About
          </div>
        </a>
        <div
          class="shopping-cart-article-holder"
        >
          <span
            class="shopping-cart-article-counter"
          />
          <div
            class="nav-links cart-svg"
          >
            <svg>
              shopping_cart.svg
            </svg>
          </div>
        </div>
        <div
          class="toggle-cart-display"
          id="cart"
        >
          <div
            class="shopping-cart-container"
          >
            <div
              class="shopping-cart-header-container"
            >
              <div
                class="shopping-cart-headline"
              >
                Shopping Cart
              </div>
              <span
                class="close-shopping-cart"
              >
                Close
              </span>
            </div>
            <div
              class="shopping-cart-article-list"
            >
              <div>
                Shopping Cart Empty
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="app-content-container"
    >
      <div
        class="home"
      >
        <div>
          Welcome to Genshin Shop!
        </div>
        <p>
           We are a shop designd to give you the best experience in online shopping that the world can offer.
        </p>
      </div>
    </div>
    <footer
      class="app-footer"
    >
      Footer Content
    </footer>
  </div>
</DocumentFragment>
`;
